pauthabielf64/pauthabielf64.rst

```diff
@@ -832,19 +834,17 @@ language to signing schema is expected to evolve over time. Even if
 the low-level ELF extensions remain constant, a change to the
 high-level language mapping may result in incompatible ELF files.
 
+A processor-specific program property type
+``GNU_PROPERTY_AARCH64_FEATURE_PAUTH`` is defined in section
+``.note.gnu.property`` [LINUX_ABI_] with a value of ``0xc0000001``.
 Every relocatable object, executable and shared library that uses the
-PAuth ABI ELF extensions must have a section named
-``.note.AARCH64-PAUTH-ABI-tag`` of type ``SHT_NOTE``. This section is
-structured as a note section as documented in SCO-ELF_.
-
-The name field (``namesz`` / ``name``) contains the string "ARM". The
-type field shall be 1, and the ``descsz`` field must be at least 16.
-The first 16 bytes of the description must contain 2 64-bit words, with
-the first 64-bit word being a platform identifier, and the second
-64-bit word being a version number for the ABI for the platform
-identified for the first word. When ``descsz`` is larger than 16 the
-remainder of the contents of desc are defined by the (platform id,
-version number).
+PAuth ABI ELF extensions must contain this property. The format of
+the data in ``pr_data`` is at least two 64-bit words, the first being
+a platform identifier, and the second being a version number specific
+to the platform identified in the first word. Consequently, the
+``pr_datasz`` field must be at least 16. When ``pr_datasz`` is larger
+than 16, the remainder of the contents of ``pr_data`` are specific
+to the (platform id, version number).
 
 This ABI does not determine the format of the platform identifier. Arm
 reserves the platform id 0 for a bare-metal platform.
```

Comment:
@smithp35 would it be reasonable to specify some format, or at least add the values in a table so that platforms can add their identifiers here as they get allocated? I.e.
|| Platform || ID ||
| Baremetal | 0 |
| | <ID 1> |
| ... | ... |

Keeping the allocations clearly in one place might help platforms avoid stepping on each other's toes.

Reply:
I was expecting the majority of holders to use short null terminated strings such as "Arm\0" as there is 64-bits to play with. This would limit the possibility of clashes. However no objections to introducing a table. With the ABI in Github it should be easy to submit pull requests to update the table.

(On holiday today, so will be a bit slow responding)

Reply:
Peter, thank you for the review. Feel free to respond after you are back from holidays :) 

How about a table like this:

|| Platform || ID (hex) || ID (human-readable) ||
| Baremetal | 0x0 | 0 |
| Platform 1 | 0x506c617431000000 | "Plat1" |

Then we can define in this ABI the rule by which two platforms are different. For example:

* The rule is that the full 64 bytes must be the same. "Arm\0\0\0\0\0" != "Arm\0Leg\0". I prefer this rule as it allows platforms to decide on the human-readable representation (string, integer, whatever) as long as the hex value is different to the rest of the platforms.

* The rule is that the field represents a null-terminated string. "Arm\0\0\0\0\0" and "Arm\0Leg\0" are the same vendor.

Defining the rule in this ABI would make it easier for tools to implement the comparison consistently for all platforms.

WDYT?